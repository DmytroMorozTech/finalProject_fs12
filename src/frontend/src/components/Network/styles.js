import { makeStyles } from '@material-ui/core/styles'

export default makeStyles(() => ({

  network: {
    marginTop: '75px'
  },

  '@media screen and (max-width: 2000px)': {
    networkLeft: {
      width: '325px'
    },

    networkMain: {
      width: '807px'
    }
  },

  '@media screen and (max-width: 1160px)': {
    networkLeft: {
      width: '300px'
    },

    networkMain: {
      width: '617px'
    }
  },

  '@media screen and (max-width: 945px)': {
    networkLeft: {
      width: '300px'
    },

    networkMain: {
      width: '427px'
    }
  },

  '@media screen and (max-width: 786px)': {
    networkLeft: {
      width: '617px'
    },

    networkMain: {
      width: '617px'
    }
  },

  '@media screen and (max-width: 645px)': {
    networkLeft: {
      minWidth: '380px',
      width: '427px'
    },

    networkMain: {
      minWidth: '380px',
      width: '427px'
    }
  },

  '@media screen and (max-width: 400px)': {
    networkLeft: {
      minWidth: '300px'
    },

    networkMain: {
      minWidth: '300px'
    }
  }
}))
