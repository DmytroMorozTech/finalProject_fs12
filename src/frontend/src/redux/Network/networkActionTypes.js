export const SAVE_INVITATIONS_FOR_ME = 'SAVE_INVITATIONS_FOR_ME'
export const SAVE_INVITATIONS_FROM_ME = 'SAVE_INVITATIONS_FROM_ME'
export const SAVE_FOLLOWED_AND_FOLLOWING = 'SAVE_FOLLOWED_AND_FOLLOWING'

export const ADD_NEW_INVITATION = 'ADD_NEW_INVITATION' // created new invitation
export const DELETE_INVITATION = 'DELETE_INVITATION' // when user ignores (declines) invitation
export const ACCEPT_INVITATION = 'ACCEPT_INVITATION'

export const SAVE_MY_CONNECTIONS = 'SAVE_MY_CONNECTIONS'
export const DELETE_CONNECTION = 'DELETE_CONNECTION'
