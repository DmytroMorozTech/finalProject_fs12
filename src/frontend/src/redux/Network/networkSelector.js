export const invitationsForMeSelector = (store) => store.network.invitationsForMe
export const invitationsFromMeSelector = (store) => store.network.invitationsFromMe
export const invitationsAreLoading = (store) => store.network.invitationsAreLoading
