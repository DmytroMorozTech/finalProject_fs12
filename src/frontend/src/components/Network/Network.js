import React from 'react'
import styles from './styles'
import NetworkMain from './NetworkMain/NetworkMain'
import Grid from '@material-ui/core/Grid'
import { Container } from '@material-ui/core'
import NetworkLeft from './NetworkLeft/NetworkLeft'

function Network () {
  const classes = styles()
  return (
    <Container className={classes.network} maxWidth={'lg'}>
      <Grid container spacing={2} alignItems="flex-start" justifyContent="center">

        <Grid item xs={12} sm={5} md={4} lg={3}>
          <NetworkLeft/>
        </Grid>

        <Grid item xs={12} sm={7} md={7} lg={8} xl={8}>
          <NetworkMain/>
        </Grid>

      </Grid>
    </Container>
  )
}

export default Network
