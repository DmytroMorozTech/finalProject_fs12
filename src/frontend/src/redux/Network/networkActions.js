import http from '../../services/httpService'
import * as actions from '../Network/networkActionTypes'
import { toast } from 'react-toastify'

export const createNewInvitationAction = (payload) => (dispatch) => {
  const {userWhomId, fullName} = payload
  dispatch({ type: actions.DELETE_POTENTIAL_CONTACT, payload: userWhomId })
  return http
    .post('/api/invitations', { ...payload })
    .then((res) => res.data)
    .then((invitationObj) => {
      dispatch({ type: actions.ADD_NEW_INVITATION, payload: invitationObj })
      toast.info(`You've just invited ${fullName} to become you Connection`)
    })
    .catch(err => {
      const errorMsg = err.response.data.message
      toast.error(errorMsg)
    })
}

export const getInvitationsForMeAction = () => (dispatch) => {
  http
    .get('/api/invitations/for_me')
    .then(res => {
      dispatch({
        type: actions.SAVE_INVITATIONS_FOR_ME,
        payload: res.data
      })
    })
    .catch(err => {
      const errorMsg = err.response.data.message
      toast.error(errorMsg)
    })
}

export const getInvitationsFromMeAction = () => (dispatch) => {
  http
    .get('/api/invitations/from_me')
    .then(res => {
      dispatch({
        type: actions.SAVE_INVITATIONS_FROM_ME,
        payload: res.data
      })
    })
    .catch(err => {
      const errorMsg = err.response.data.message
      toast.error(errorMsg)
    })
}

export const getMyConnectionsAction = () => (dispatch) => {
  http
    .get('/api/users/connections')
    .then(res => {
      dispatch({
        type: actions.SAVE_MY_CONNECTIONS,
        payload: res.data
      })
    })
    .catch(err => {
      const errorMsg = err.response.data.message
      toast.error(errorMsg)
    })
}

export const deleteConnectionAction = (activeUserId, userWhomId) => (dispatch) => {
  return http
    .delete(`/api/connections/${activeUserId}/${userWhomId}`)
    .then((res) => res.status)
    .then((status) => {
      if (status === 200 || status === 204) {
        dispatch({ type: actions.DELETE_CONNECTION, payload: userWhomId })
      }
    })
    .catch(err => {
      const errorMsg = err.response.data.message
      toast.error(errorMsg)
    })
}

export const deleteInvitationAction = (id) => (dispatch) => {
  return http
    .delete(`/api/invitations/${id}`)
    .then((res) => res.status)
    .then((status) => {
      if (status === 200 || status === 204) {
        dispatch({ type: actions.DELETE_INVITATION, payload: id })
      }
    })
    .catch(err => {
      const errorMsg = err.response.data.message
      toast.error(errorMsg)
    })
}

export const acceptInvitationAction = (invitationId) => (dispatch) => {
  return http
    .put(`/api/invitations/accept/${invitationId}`)
    .then((res) => res.status)
    .then((status) => {
      if (status === 200 || status === 204) {
        toast.info(`Invitation with id: ${invitationId} was accepted`)
        dispatch({ type: actions.DELETE_INVITATION, payload: invitationId })
      }
    })
    .catch(err => {
      const errorMsg = err.response.data.message
      toast.error(errorMsg)
    })
}

export const getUsersFollowingAction = () => (dispatch) => {
  http
    .get('/api/users/following')
    .then(res => {
      dispatch({
        type: actions.SAVE_USERS_FOLLOWING,
        payload: res.data
      })
    })
    .catch(err => {
      const errorMsg = err.response.data.message
      toast.error(errorMsg)
    })
}

export const getUsersFollowedAction = () => (dispatch) => {
  http
    .get('/api/users/followed')
    .then(res => {
      dispatch({
        type: actions.SAVE_USERS_FOLLOWED,
        payload: res.data
      })
    })
    .catch(err => {
      const errorMsg = err.response.data.message
      toast.error(errorMsg)
    })
}

export const toggleUserFollowedAction = (userId, setIsFollowed, isFollowed) => () => {
  http
    .put(`/api/users/toggle_follow_user/${userId}`)
    .then(res => res.status)
    .then(status => {
      if (status === 200 || status === 204) {
        setIsFollowed(!isFollowed)
      }
    })
    .catch(err => {
      const errorMsg = err.response.data.message
      toast.error(errorMsg)
    })
}

export const getPotentialContacts = () => (dispatch) => {
  http
    .get('/api/users/potential_contacts')
    .then(res => {
      dispatch({
        type: actions.SAVE_POTENTIAL_CONTACTS,
        payload: res.data
      })
    })
    .catch(err => {
      const errorMsg = err.response.data.message
      toast.error(errorMsg)
    })
}
