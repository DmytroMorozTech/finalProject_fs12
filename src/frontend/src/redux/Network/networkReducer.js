import * as actions from '../Network/networkActionTypes'

const initialState = {
  invitationsForMe: [],
  invitationsFromMe: [],
  usersFollowingMe: [],
  usersFollowedByMe: [],
  connections: []
}

const networkReducer = (state = initialState, action) => {
  switch (action.type) {
    case actions.ADD_NEW_INVITATION:
      return { ...state, invitationsFromMe: [ ...state.invitationsFromMe, action.payload ] }

    case actions.SAVE_INVITATIONS_FOR_ME:
      return { ...state, invitationsForMe: [...action.payload] }

    case actions.SAVE_INVITATIONS_FROM_ME:
      return { ...state, invitationsFromMe: [...action.payload] }

    case actions.DELETE_INVITATION:
      const invitationId = action.payload
      const invitationsForMeUpdated = state.invitationsForMe.filter(inv => inv.id !== invitationId)
      const invitationsFromMeUpdated = state.invitationsFromMe.filter(inv => inv.id !== invitationId)
      return { ...state,
        invitationsForMe: [...invitationsForMeUpdated],
        invitationsFromMe: [...invitationsFromMeUpdated]
      }

    case actions.DELETE_CONNECTION:
      const connectedUserId = action.payload
      const filteredConnections = state.connections.filter(connection => connection.id !== connectedUserId)
      return { ...state, connections: [...filteredConnections]
      }

    case actions.SAVE_MY_CONNECTIONS:
      return { ...state, connections: [...action.payload] }

    case actions.SAVE_USERS_FOLLOWING:
      return { ...state, usersFollowingMe: [...action.payload] }

    case actions.SAVE_USERS_FOLLOWED:
      return { ...state, usersFollowedByMe: [...action.payload] }

    default: {
      return state
    }
  }
}

export default networkReducer
