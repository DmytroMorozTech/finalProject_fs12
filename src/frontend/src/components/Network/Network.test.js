import React from 'react'
import Network from './Network'
import { shallow } from 'enzyme'
import * as redux from 'react-redux'

const setUp = () => shallow(<Network />)

describe('should render Network component', () => {
  let spyOnUseSelector
  let spyOnUseDispatch
  let mockDispatch

  let component

  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector')
    spyOnUseSelector.mockReturnValue(false)
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    component = setUp()
  })

  afterEach(() => {
    jest.restoreAllMocks()
  })

  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
