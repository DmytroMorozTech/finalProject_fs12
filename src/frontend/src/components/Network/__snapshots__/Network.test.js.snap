// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Network component to match snapshot 1`] = `
<WithStyles(ForwardRef(Container))
  className="makeStyles-network-1"
  maxWidth="lg"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    justifyContent="center"
    spacing={4}
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-networkLeft-2"
      item={true}
    >
      <NetworkLeft />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-networkMain-3"
      item={true}
    >
      <NetworkMain
        invitations={false}
        potentialContacts={false}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Container))>
`;
